app/views/articles/show.html.erb

```diff
@@ -13,5 +13,23 @@
   <%= @article.status %>
 </p>
 
+<% if @article.status == 'draft' %>
+  <%= form_with(model: @article) do |form| %>
```

Comment:
How about add different route for updating article status to draft/published? In this case you don't need form with checkbox